### DISCUSSION TITLE
useHead script on component or page

### DISCUSSION THREAD

**Discussion by amos81** [nuxt/nuxt#23132] [Questions]
I have a form component that I want to use third party JS and CSS for. I've tried useHead link and script in the component's script setup but when I navigate to a page with that component the JS does not load. If I reload the page, or restart the app on that page, it works. I've also tried useServerHead and got the same result. Moving or copying useHead to the page's script setup also resulted in the same behaviour.

Is there anything I can try beyond just moving the JS and CSS into the app:head in the nuxt config?

Code in component:
```
useHead({
	script: [ { src: 'forms.js', defer: true } ],
	link: [ { rel: 'stylesheet', href: 'forms.css' } ]
})
```

**Comment by xjccc** [comment]
try use in layouts or [head](https://nuxt.com/docs/api/configuration/nuxt-config#head)

**Comment by HiddenUndefined** [comment]
If you still need a help with your problem, maybe you can create stackblitz with example of your case?

I try do your case in my tiny app and script and styles are loaded after component mounting and work if it unmount and mounted again.

Maybe you help this link: [Nuxt 3: API – useHead](https://nuxt.com/docs/api/composables/use-head#type).
